src/search/plugins/raw_registry.cc

```diff
@@ -118,56 +118,68 @@ SubcategoryPlugins RawRegistry::collect_subcategory_plugins(
-        key_occurrences[key]++;
-        features[key] = move(feature);
+        string feature_key = feature->get_key();
+        feature_key_occurrences[feature_key]++;
+        features[feature_key] = move(feature);
     }
 
-    // Check that keys are unique
-    for (const auto &pair : key_occurrences) {
-        const string &key = pair.first;
+    // Check that feature_keys are unique
+    for (const auto &pair : feature_key_occurrences) {
+        const string &feature_key = pair.first;
         int occurrences = pair.second;
         if (occurrences > 1) {
             errors.push_back(
                 to_string(occurrences) + " Features are defined for the key '" +
-                key + "'.");
+                feature_key + "'.");
         }
     }
 
     // Check that all subcategories used in features are defined
     for (const auto &item : features) {
-        const string &key = item.first;
+        const string &feature_key = item.first;
         const Feature &feature = *item.second;
         string subcategory = feature.get_subcategory();
 
         if (!subcategory.empty() && !subcategory_plugins.count(subcategory)) {
             const Type &type = feature.get_type();
             errors.push_back(
                 "Missing SubcategoryPlugin '" + subcategory + "' for Plugin '" +
-                key + "' of type " + type.name());
+                feature_key + "' of type " + type.name());
         }
     }
 
     // Check that all types used in features are defined
     unordered_set<type_index> missing_types;
     for (const auto &item : features) {
-        const string &key = item.first;
+        const string &feature_key = item.first;
         const Feature &feature = *item.second;
 
         const Type &type = feature.get_type();
         if (type == TypeRegistry::NO_TYPE) {
             errors.push_back(
-                "Missing Plugin for type of feature '" + key + "'.");
+                "Missing Plugin for type of feature '" + feature_key + "'.");
         }
 
+        unordered_map<string, int> parameter_occurrences;
         for (const ArgumentInfo &arg_info : feature.get_arguments()) {
             if (arg_info.type == TypeRegistry::NO_TYPE) {
                 errors.push_back(
-                    "Missing Plugin for type of argument '" + arg_info.key
-                    + "' of feature '" + key + "'.");
+                    "Missing Plugin for type of parameter '" + arg_info.key
+                    + "' of feature '" + feature_key + "'.");
+            }
+            ++parameter_occurrences[arg_info.key];
+        }
+        // Check that parameters are unique
+        for (const auto &pair : parameter_occurrences) {
+            const string &parameter = pair.first;
+            int parameter_occurrence = pair.second;
+            if (parameter_occurrence > 1) {
+                errors.push_back(
+                    "The Parameter '" + parameter + "' in '" + feature_key + "' is defined " +
```

Comment:
Before you merge, can you change "Parameter" to "parameter" in the message? (The existing messages can also be criticized for the same kind of grammatical mistake, but we can argue that we think of "Plugin" as a technical proper noun in this context. For "parameter" we can't.)